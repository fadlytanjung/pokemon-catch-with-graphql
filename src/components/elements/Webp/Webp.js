import React from 'react';
import PropTypes from 'prop-types';

export default function Webp(props) {
  const { alt, className, name, style } = props;

  return (
    <picture className={className} style={style}>
      <source srcSet={`/assets/${name}.webp`} type="image/webp" />
      <source srcSet={`/assets/${name}.png`} type="image/png" />
      <img alt={alt} src={`/assets/${name}.png`} />
    </picture>
  );
}

Webp.defaultProps = {
  alt: '',
  className: '',
  name: '',
  style: {},
};

Webp.propTypes = {
  alt: PropTypes.string,
  className: PropTypes.string,
  name: PropTypes.string,
  style: PropTypes.object,
};
