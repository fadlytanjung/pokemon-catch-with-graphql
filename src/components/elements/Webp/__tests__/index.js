import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Webp from '../Webp';

describe('src/components/elements/Webp', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Webp />);
    expect(tree).toMatchSnapshot();
  });
});
