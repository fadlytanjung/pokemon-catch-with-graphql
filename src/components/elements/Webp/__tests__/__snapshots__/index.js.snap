// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/Webp render 1`] = `
<picture
  className=""
  style={Object {}}
>
  <source
    srcSet="/assets/.webp"
    type="image/webp"
  />
  <source
    srcSet="/assets/.png"
    type="image/png"
  />
  <img
    alt=""
    src="/assets/.png"
  />
</picture>
`;
